core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -17,58 +17,40 @@
 
 package kafka.server
 
-import java.util.Collections
-import java.util.Optional
-
 import kafka.api._
 import kafka.cluster.BrokerEndPoint
 import kafka.log.{LeaderOffsetIncremented, LogAppendInfo}
-import kafka.server.AbstractFetcherThread.ReplicaFetch
-import kafka.server.AbstractFetcherThread.ResultWithPartitions
+import kafka.server.AbstractFetcherThread.{ReplicaFetch, ResultWithPartitions}
 import kafka.utils.Implicits._
 import org.apache.kafka.clients.FetchSessionHandler
-import org.apache.kafka.common.{TopicPartition, Uuid}
 import org.apache.kafka.common.errors.KafkaStorageException
-import org.apache.kafka.common.message.ListOffsetsRequestData.{ListOffsetsPartition, ListOffsetsTopic}
-import org.apache.kafka.common.message.OffsetForLeaderEpochRequestData.OffsetForLeaderTopic
-import org.apache.kafka.common.message.OffsetForLeaderEpochRequestData.OffsetForLeaderTopicCollection
-import org.apache.kafka.common.message.OffsetForLeaderEpochResponseData.EpochEndOffset
-import org.apache.kafka.common.metrics.Metrics
-import org.apache.kafka.common.protocol.Errors
 import org.apache.kafka.common.record.MemoryRecords
 import org.apache.kafka.common.requests._
-import org.apache.kafka.common.utils.{LogContext, Time}
+import org.apache.kafka.common.{TopicPartition, Uuid}
 
-import scala.jdk.CollectionConverters._
+import java.util.Optional
 import scala.collection.{Map, mutable}
 import scala.compat.java8.OptionConverters._
 
 class ReplicaFetcherThread(name: String,
-                           fetcherId: Int,
+                           leader: LeaderEndPoint,
                            sourceBroker: BrokerEndPoint,
                            brokerConfig: KafkaConfig,
                            failedPartitions: FailedPartitions,
                            replicaMgr: ReplicaManager,
-                           metrics: Metrics,
-                           time: Time,
                            quota: ReplicaQuota,
-                           leaderEndpointBlockingSend: Option[BlockingSend] = None)
+                           private[server] val fetchSessionHandler: FetchSessionHandler,
```

Comment:
It's a bit weird to have fetchSessionHandler in two places, ReplicaFetcherThread and RemoteLeaderEndPoint. The only reason that we need fetchSessionHandler in ReplicaFetcherThread is for buildFetch(). Could we move buildFetch() to LeaderEndPoint too?